airflow/providers/cncf/kubernetes/hooks/kubernetes.py

```diff
@@ -232,9 +234,12 @@ def get_conn(self) -> Any:
 
         if in_cluster:
             self.log.debug("loading kube_config from: in_cluster configuration")
+            self._is_in_cluster = True
             config.load_incluster_config()
             return client.ApiClient()
 
+        self._is_in_cluster = False
```

Comment:
it's a tiny bit odd to  set it false  here when  you might set it to  true again later. 

even though it's more code, it would probably be a tiny  bit better to set this when we know, i.e. closer to the return  statements.  e.g. right before  250 and 262 and conditionally in _get_default_client

until we _know_, it's probably  better to leave it  not set.

Reply:
I don't love the sprawl, but it does make it easier to test as well. Done.